pandas/core/dtypes/base.py

```diff
@@ -452,4 +451,4 @@ def find(
         return None
 
 
-registry = Registry()
+registry: Registry = Registry()
```

Comment:
why is this needed?

Reply:
I think that the `pylance` type checker wanted that.  I need to investigate it.

Reply:
Would it work to use stricter flags like
```
disallow_incomplete_defs = true
disallow_untyped_defs = true
```
to discover missing annotations, so then we don't have to add type hints which are redundant for `mypy`?

Reply:
we may have already have issues for these. mypy has flags to activate at milestones in the typing journey and mypy also generates reports. (we don't really need to use pyright)

differences of opinion will arise when the motivations are typing public apis vs in-line typing for code consistency/robustness.

from https://blog.zulip.com/2016/10/13/static-types-in-python-oh-mypy/

"For Zulip, getting clean --check-untyped-defs output took me about 2 days of hard work" - it too us over a year.

Fully annotating a large codebase:
Phase 1: Annotate the core library - we need stubs for the cython files and to get the numpy types in play[1].
Phase 2: Annotate the bulk of the codebase. - 
Phase 3: Get to 100%. - use --disallow-untyped-defs to lockdown progress

so there is much todo to complete phase 1 before we can be confident in the typing of the public api.

[1] https://github.com/pandas-dev/pandas/pull/39501#discussion_r590339840 would be picked up by mypy, i.e. trying to pass `None` through to numpy.searchsorted.

Reply:
Actually, I think the right solution here is to make `registry` be `_registry`, because it is not part of the public interface.  What `pyright --verifytypes` does is verify that anything that is in the public interface is properly typed.  I will make that change.